sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Relation.scala

```diff
@@ -106,7 +106,7 @@ case class StreamingDataSourceV2Relation(
 
 object DataSourceV2Relation {
   private implicit class SourceHelpers(source: DataSourceV2) {
-    def asReadSupport: ReadSupport = {
+    private def asReadSupport: ReadSupport = {
```

Comment:
Minor: these are effectively private because `SourceHelpers` is private. If we were to move `SourceHelpers` or make it public to some other part of Spark, we would have to revert these changes. So I think it is best to rely on the visibility of `SourceHelpers` instead of making these private.

Reply:
makes sense